components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -99,13 +105,17 @@ abstract class AbstractFetchDownloadService : Service() {
 
                 when (intent.action) {
                     ACTION_PAUSE -> {
-                        currentDownloadJobState.status = DownloadJobStatus.PAUSED
+                        synchronized(context) {
+                            currentDownloadJobState.status = DownloadJobStatus.PAUSED
+                        }
```

Comment:
Now that we synchronize you can remove the `@Volatile` from `DownloadJobState`. Instead you could try adding `@GuardedBy("context")` to the `status` property to denote that this needs to be sychronized on `context` when accessed/modified. That itself doesn't do anything but hopefully the IDE (or lint) will show a warning if you don't do that.

Reply:
Ooo didn't know about that keyword. Thanks!